SDWebImage/Core/SDImageCache.m

```diff
@@ -440,8 +440,7 @@ - (nullable UIImage *)imageFromCacheForKey:(nullable NSString *)key options:(SDI
     if (image) {
         if (options & SDImageCacheDecodeFirstFrameOnly) {
             // Ensure static image
-            Class animatedImageClass = image.class;
-            if (image.sd_isAnimated || ([animatedImageClass isSubclassOfClass:[UIImage class]] && [animatedImageClass conformsToProtocol:@protocol(SDAnimatedImage)])) {
```

Comment:
This check is no longer needed. This exist is because of history code that `SDAnimatedImage` and `YYImage` thing does not implements the `sd_isAnimatedImage` method